lib/kafo/exit_handler.rb

```diff
@@ -26,6 +26,7 @@ def error_codes
     def exit(code, &block)
       @exit_code = translate_exit_code(code)
       block.call if block
+      KafoConfigure.hooking.execute(:pre_exit, log_stage: false)
```

Comment:
I must say that I dislike the hardcoded dependency on hooking here (just as I don't like the use of `KafoConfigure.logger`). IMHO it would be cleaner if `hooking` was passed in via the constructor as an optional parameter.